app/src/main/java/io/intrepid/contest/screens/entrysubmission/entryname/EntryNameActivity.java

```diff
@@ -0,0 +1,75 @@
+package io.intrepid.contest.screens.entrysubmission.entryname;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.widget.Button;
+import android.widget.TextView;
+
+import butterknife.BindView;
+import butterknife.OnFocusChange;
+import butterknife.OnTextChanged;
+import io.intrepid.contest.R;
+import io.intrepid.contest.base.BaseMvpActivity;
+import io.intrepid.contest.base.PresenterConfiguration;
+
+import static io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.Presenter;
+import static io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.View;
+
+public class EntryNameActivity extends BaseMvpActivity<Presenter> implements View {
+
+    @BindView(R.id.contestant_welcome_text_view)
+    TextView contestantWelcomeTextView;
+    @BindView(R.id.entry_name_next_button)
+    Button entryNameNextButton;
+
+    @NonNull
+    @Override
+    public Presenter createPresenter(PresenterConfiguration configuration) {
+        return new EntryNamePresenter(this, configuration);
+    }
+
+    @Override
+    protected int getLayoutResourceId() {
+        return R.layout.activity_entry_name;
+    }
+
+    @Override
+    protected void onViewCreated(Bundle savedInstanceState) {
+        super.onViewCreated(savedInstanceState);
+    }
+
+    @Override
+    public void onContestNameRetrieved(String contestName) {
+        contestantWelcomeTextView.setText(getResources().getString(R.string.contestant_welcome_message, contestName));
+    }
+
+    @OnFocusChange(R.id.entry_name_edit_text)
+    public void onEntryNameTextViewFocusChanged(boolean isFocused) {
+        presenter.onEntryNameTextViewFocusChanged(isFocused);
+    }
+
+    @OnTextChanged(R.id.entry_name_edit_text)
+    public void onEntryNameTextViewTextChanged(CharSequence newText, int start, int before, int count) {
+        presenter.onEntryNameTextViewTextChanged(newText);
+    }
+
+    @Override
+    public void showEntryNameNextButton() {
+        entryNameNextButton.setVisibility(android.view.View.VISIBLE);
```

Comment:
Its a bit of a stylistic change but you can possibly simplify the number of methods by passing in a boolean value. This would become:

showHideEntryButton(boolean show){
    int visibility = (show) ? View.Visible : View.INVISIBLE;
    entryNameNextButton.setVisibility(visibility)
}

It cuts down a bit on the number of function. I have seen people do both ways so I'll leave the choice up to what you two feel is clearer.